modules/ref_arch_setup/tasks/download_pe_tarball.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+# Download PE Tarball:
+# This is a script-based task as the target node may not have Ruby installed.
+# Provide the URL and optionally an existing destination directory (the default is /tmp/ref_arch_setup).
+# Run the task as per the following examples:
+#
+# Localhost with default destination -
+# bolt task run ref_arch_setup::download_pe_tarball url=http://test.net/test.gz --modulepath ./modules --nodes localhost
+#
+# Remote node as root user with specified destination -
+# bolt task run ref_arch_setup::download_pe_tarball url=http://test.net/test.gz destination=/tmp/ras --modulepath ./modules --nodes tjsss144r6ouvhc --user root
+
+URL=$PT_url
+DESTINATION=${PT_destination:-/tmp/ref_arch_setup}
+FILENAME=$(basename $PT_url)
+DESTINATION_PATH=$DESTINATION/$FILENAME
+
+echo URL: $URL
+echo Destination: $DESTINATION
+echo Filename: $FILENAME
+echo
+
+if [[ $URL =~ https?://.*gz ]] ; then
+  echo "URL is valid; proceeding"
+else
+  echo "Invalid URL (must be http or https with a .gz extension); exiting"
+  exit 1
+fi
+
+if [ -d "$DESTINATION" ]; then
```

Comment:
I think we may want to create the directory for them if it does not already exist, otherwise we would require them to ssh into the master separately to create a directory if they want to use any non-standard directory.